scripts/deploymentextensions/enablemobileapi/install.sh

```diff
@@ -0,0 +1,394 @@
+    azure_subscription_parameters="--azure-subscription-id ""${azure_subscription_id}"" --azure-resource-group ""${azure_resource_group}"""
+    misc_parameters="--cluster-admin-email ""${cluster_admin_email}"" --target-user ""${target_user}"" --remote"
+
+    # conditionally enable debug mode over the remote session
+    if [[ $debug_mode == 1 ]];
+    then
+        misc_parameters+=" --debug"
+    fi
+
+    remote_command="sudo bash ~/install.sh ${repository_parameters} ${aad_parameters} ${azure_subscription_parameters} ${misc_parameters}"
+
+    # run the remote command
+    ssh "${remote_execution_target_user}@${remote_execution_server_target}" $remote_command
+    exit_on_error "Could not execute the tools installer on the remote target: ${remote_execution_server_target} from '${HOSTNAME}' !" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+}
+
+authenticate-azureuser()
+{
+    # this call requires azure cli2
+
+    # login
+    results=`az login -u $aad_webclient_id --service-principal --tenant $aad_tenant_id -p $aad_webclient_appkey --output json`
+    exit_on_error "Could not login to azure with the provided service principal credential from '${HOSTNAME}' !" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+
+    # select the appropriate subscription to set the execution context
+    az account set --subscription "${azure_subscription_id}"
+    exit_on_error "Could not set the azure subscription context to ${azure_subscription_id} from '${HOSTNAME}' !" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+}
+
+validate_args()
+{
+    log "Validating arguments"
+
+    # target user
+    if [[ -z $target_user ]]; 
+    then
+        log "You must specify a user account to use for SSH to remote servers"
+        exit $error_mobilerestapi_update_failed
+    fi
+
+    if [[ -z $aad_webclient_id ]] || [[ -z $aad_webclient_appkey ]] || [[ -z $aad_tenant_id ]]; 
+    then
+        log "You must specify a valid credential for your azure subscription authentication. You need the AAD Webclient Id, Webclient Key and AAD tenant id"
+        exit $error_mobilerestapi_update_failed
+    fi
+
+    if [[ -z $azure_subscription_id ]] || [[ -z $azure_resource_group ]]; 
+    then
+        log "You must specify a valid Azure Subscription Id and Resource group representing the stamp cluster."
+        exit $error_mobilerestapi_update_failed
+    fi
+
+    if [[ -z $cloud ]]; 
+    then
+        log "You must specify a target cloud for this deployment to ensure it cloud configuration gets updated"
+        exit $error_mobilerestapi_update_failed
+    fi
+
+    log "Completed argument validation successfully"
+}
+
+###############################################
+# START CORE EXECUTION
```

Comment:
This has been considered and mentioned in the original notification preceding this PR.